packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart

```diff
@@ -89,22 +91,25 @@ class _MemoryChartPaneState extends State<MemoryChartPane>
           _hideHover();
         }
         final tapLocation = widget.chartController.event.tapLocation.value;
-        if (tapLocation?.tapDownDetails != null) {
-          final tapData = tapLocation!;
-          final index = tapData.index;
-          final timestamp = tapData.timestamp!;
-
-          final copied = TapLocation.copy(tapLocation);
-          widget.chartController.vm.tapLocation.value = copied;
-          widget.chartController.android.tapLocation.value = copied;
-
-          final allValues = ChartsValues(controller, index, timestamp);
-          _showHover(
-            context,
-            allValues,
-            tapData.tapDownDetails!.globalPosition,
-          );
-        }
+        if (tapLocation == null) return;
+        final tapDownDetails = tapLocation.tapDownDetails;
+        if (tapDownDetails == null) return;
```

Comment:
shorten to: `if (tapLocation?.tapDownDetails == null) return;`

Reply:
FWIW, I think I'd rather stick with `if (tapLocation?.tapDownDetails == null) return;` and then change line 98 to `final copied = TapLocation.copy(tapLocation!);` since that will only introduce one extra null check. Then line 111 could just be `tapLocation.tapDownDetails!.globalPosition`.

Reply:
This PR is to fix null reference error. It is not clear where is it originated. So, I prefer to avoid any bungs in this code.